src/Functions/tokens.cpp

```diff
@@ -18,12 +18,75 @@ namespace ErrorCodes
     extern const int ILLEGAL_COLUMN;
 }
 
+namespace
+{
+template <typename T>
+std::optional<T> getArgument(const ColumnsWithTypeAndName & arguments, size_t argument_index, std::string_view function_name = "")
```

Comment:
Minor:

Considering all the conditions+the `static_assert`. Isn't it simpler just write an unimplemented generic template definition and 3 specializations for `string_view`, `Uint64` and `vector<String>` without the `if constexpr`? Mainly because there is almost no code overlap between conditions.

In that way it will be easier to extend with a concept in the future for example to support Uint32 or string_views if needed. See comment bellow.